systemtest/src/main/java/io/strimzi/systemtest/utils/specific/CruiseControlUtils.java

```diff
@@ -79,33 +79,54 @@ public static void verifyCruiseControlMetricReporterConfigurationInKafkaConfigMa
+
+                if (modelTrainingSamples != null && partitionsMetricsSamples != null) {
+                    boolean hasTopicCorrectPartitionsCount =
+                            modelTrainingSamples.getSpec().getPartitions() == numberOfPartitionsSamplesTopic &&
+                            partitionsMetricsSamples.getSpec().getPartitions() == numberOfPartitionsSamplesTopic;
+
+                    boolean hasTopicCorrectReplicasCount =
+                            modelTrainingSamples.getSpec().getReplicas() == numberOfReplicasSamplesTopic &&
+                            partitionsMetricsSamples.getSpec().getReplicas() == numberOfReplicasSamplesTopic;
+
+                    return hasTopicCorrectPartitionsCount && hasTopicCorrectReplicasCount;
+                }
+                LOGGER.debug("One of the samples {}, {} topics are not present", CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC, CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC);
+                return false;
+            });
+    }
+
+    public static void verifyThatKafkaCruiseControlMetricReporterTopicIsPresent(long timeout) {
+        final int numberOfPartitionsMetricTopic = 1;
+        final int numberOfReplicasMetricTopic = 1;
 
         TestUtils.waitFor("Verify that kafka contains cruise control topics with related configuration.",
-            Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_CRUISE_CONTROL_TIMEOUT, () -> {
+            Constants.GLOBAL_POLL_INTERVAL, timeout, () -> {
+                KafkaTopic metrics = KafkaTopicResource.kafkaTopicClient().inNamespace(kubeClient().getNamespace()).withName(CRUISE_CONTROL_METRICS_TOPIC).get();
 
                 boolean hasTopicCorrectPartitionsCount =
-                    metrics.getSpec().getPartitions() == numberOfPartitionsMetricTopic &&
-                    modelTrainingSamples.getSpec().getPartitions() == numberOfPartitionsSamplesTopic &&
-                    partitionsMetricsSamples.getSpec().getPartitions() == numberOfPartitionsSamplesTopic;
+                    metrics.getSpec().getPartitions() == numberOfPartitionsMetricTopic;
 
                 boolean hasTopicCorrectReplicasCount =
-                    metrics.getSpec().getReplicas() == numberOfReplicasMetricTopic &&
-                    modelTrainingSamples.getSpec().getReplicas() == numberOfReplicasSamplesTopic &&
-                    partitionsMetricsSamples.getSpec().getReplicas() == numberOfReplicasSamplesTopic;
+                    metrics.getSpec().getReplicas() == numberOfReplicasMetricTopic;
 
                 return hasTopicCorrectPartitionsCount && hasTopicCorrectReplicasCount;
             });
     }
 
+    public static void verifyThatCruiseControlTopicsArePresent() {
+        verifyThatCruiseControlSamplesTopicsArePresent(Constants.GLOBAL_CRUISE_CONTROL_TIMEOUT);
+        verifyThatKafkaCruiseControlMetricReporterTopicIsPresent(Constants.GLOBAL_CRUISE_CONTROL_TIMEOUT);
```

Comment:
verifyThatCruiseControlSamplesTopicsArePresent can only be true if verifyThatKafkaCruiseControlMetricReporterTopicIsPresent is true,  for that reason I think verifyThatKafkaCruiseControlMetricReporterTopicIsPresent should come first so that we can rule out verifyThatKafkaCruiseControlMetricReporterTopicIsPresent=false being the cause of verifyThatCruiseControlSamplesTopicsArePresent being false

Reply:
Good, changes @kyguy